homeassistant/components/dsmr/config_flow.py

```diff
@@ -55,40 +54,40 @@ def __init__(
         self._port = port
         self._dsmr_version = dsmr_version
         self._protocol = protocol
-        self._telegram: dict[str, DSMRObject] = {}
-        self._equipment_identifier = obis_ref.EQUIPMENT_IDENTIFIER
+        self._telegram: Telegram = None
+        self._equipment_identifier = "EQUIPMENT_IDENTIFIER"
         if dsmr_version == "5B":
-            self._equipment_identifier = obis_ref.BELGIUM_EQUIPMENT_IDENTIFIER
+            self._equipment_identifier = "BELGIUM_EQUIPMENT_IDENTIFIER"
         if dsmr_version == "5L":
-            self._equipment_identifier = obis_ref.LUXEMBOURG_EQUIPMENT_IDENTIFIER
+            self._equipment_identifier = "LUXEMBOURG_EQUIPMENT_IDENTIFIER"
         if dsmr_version == "Q3D":
-            self._equipment_identifier = obis_ref.Q3D_EQUIPMENT_IDENTIFIER
+            self._equipment_identifier = "Q3D_EQUIPMENT_IDENTIFIER"
 
     def equipment_identifier(self) -> str | None:
         """Equipment identifier."""
-        if self._equipment_identifier in self._telegram:
-            dsmr_object = self._telegram[self._equipment_identifier]
+        if hasattr(self._telegram, self._equipment_identifier):
+            dsmr_object = getattr(self._telegram, self._equipment_identifier)
```

Comment:
@frenck : but here for example you still need to use getattr

Reply:
Not really, we determine the identifier by the version. That decision can be made here and the object accessed directly.